File: 11 two Sum/two sum.py
def two_sum(nums,target):
    lookup={}
    for i in range(len(nums)-1):
        need=target-nums[i]
        if need in lookup:
            return [lookup[need],i]
        lookup[nums[i]]=i

def two_sum1(nums,target):
    left=0
    right=len(nums)-1
    while left<right:
        csum=nums[left]+nums[right]
        if csum==target:
           return [left,right]
        elif csum>target:
            right=right-1
        else:
            left=left+1

print(two_sum1([2,7,9,11],9))

File: 11 two Sum/twosum.java
import java.util.*;
public class twosum{
    public static int[] TwoSum(int[] nums, int target){
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int num = target - nums[i];
            if(map.containsKey(num)){
                return new int[] {map.get(num),i};
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String []args){
        int list[]={2,7,9,11};
        int target=9;
        int l[]=TwoSum(list, target);
        System.out.println(Arrays.toString(l));
    }
}

